lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -109,6 +112,73 @@ def get(self, request, course_id):
         return Response(get_course(request, course_key))
 
 
+class CourseUsersView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Fetch statistics about a user's activity in a course.
+
+    **Example Requests**:
+
+        GET /api/discussion/v1/courses/course-v1:ExampleX+Subject101+2015/activity_stats?order_by=activity
+
+    **Response Values**:
+
+
+    **Example Response**
+    ```json
+    {
+        "pagination": {
+            "count": 3,
+            "next": null,
+            "num_pages": 1,
+            "previous": null
+        },
+        "results": [
+            {
+                "active_flags": 3,
+                "inactive_flags": 0,
+                "replies": 13,
+                "responses": 21,
+                "threads": 32,
+                "username": "edx"
+            },
+            {
+                "active_flags": 1,
+                "inactive_flags": 0,
+                "replies": 6,
+                "responses": 8,
+                "threads": 13,
+                "username": "honor"
+            },
+            ...
+        ]
+    }
+    ```
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
```

Comment:
is those all permissions we'd like at the moment? given the API is in the testing phase, would you like to add any additional perm classes to restrict it from public access?

Reply:
These are authentication classes, they don't represent the permissions, the permissions classes are one line below this and are limmited to admins.

Reply:
ah sorry about that, I missed those. thanks. resolving.